### PR TITLE
Fix bug that auto batch size doesn't consider distributed training

### PR DESCRIPTION
**Pull request by eunwoosh**
### Summary
There is an edge case that each result of batch size try of processes are different.
This PR changes it to aggregate results and broadcasts same results to all processes.

### Checklist

- [x] I have added unit tests to cover my changes.​
- [ ] I have added integration tests to cover my changes.​
- [ ] I have added e2e tests for validation.
- [ ] I have added the description of my changes into CHANGELOG in my target branch (e.g., [CHANGELOG](https://github.com/openvinotoolkit/training_extensions/blob/develop/CHANGELOG.md) in develop).​
- [ ] I have updated the documentation in my target branch accordingly (e.g., [documentation](https://github.com/openvinotoolkit/training_extensions/tree/develop/docs) in develop).
- [ ] I have [linked related issues](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword).

### License

- [ ] I submit _my code changes_ under the same [Apache License](https://github.com/openvinotoolkit/training_extensions/blob/develop/LICENSE) that covers the project.
      Feel free to contact the maintainers if that's a concern.
- [ ] I have updated the license header for each file (see an example below).

```python
# Copyright (C) 2023 Intel Corporation
# SPDX-License-Identifier: Apache-2.0
```

### CODE REVIEW

**Comment by yunchu** [comment]
@eunwoosh could you re-target this PR to the `releases/1.5.0`?

**Review by yunchu — DISMISSED** [review]

File: src/otx/algorithms/common/adapters/torch/utils/bs_search_algo.py [hunk 1]
```diff
@@ -51,7 +52,31 @@ def _try_batch_size(self, bs: int) -> Tuple[bool, int]:
             else:
                 raise e
 
-        max_memory_allocated = torch.cuda.max_memory_allocated(device=None)
+        max_memory_allocated = torch.cuda.max_memory_reserved(device=None)
```
Comment (sungmanc) [inline src/otx/algorithms/common/adapters/torch/utils/bs_search_algo.py]:
```suggestion
        max_memory_reserved = torch.cuda.max_memory_reserved(device=None)
```
Changing the variable name is perferable

**Review by sungmanc — DISMISSED** [review]

**Review by jaegukhyun — COMMENTED** [review]
Could you add unit tests for this change?

File: src/otx/algorithms/common/adapters/torch/utils/bs_search_algo.py (hunk 1 above)
Comment (eunwoosh) [reply src/otx/algorithms/common/adapters/torch/utils/bs_search_algo.py]: all names are changed at [3c88d39](https://github.com/openvinotoolkit/training_extensions/pull/2533/commits/3c88d3930cea82a0c5645557f31675f064ef5e12)

**Review by sungmanc — APPROVED** [review]

**Review by jaegukhyun — APPROVED** [review]

**Review by goodsong81 — APPROVED** [review]
Nice work, thanks!
